pkg/apis/serving/v1beta1/revision_defaults.go

```diff
@@ -36,6 +36,13 @@ func (rts *RevisionTemplateSpec) SetDefaults(ctx context.Context) {
 
 // SetDefaults implements apis.Defaultable
 func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
+	// More than one container is not allowed for now. Do not set default
```

Comment:
While this works, it might be more appropriate to instead make it so that the code below sets defaults for **all** containers passed in through `rs.PodSpec.Containers[0]`. 

This change creates coupling between defaulting and validation. If we have the defaulter just apply defaults to the array of containers passed in we keep the separation of concerns between the defaulter and the validator.

WDYT?

Reply:
👍 I was debating to post a similar thing.